enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,641 @@
+ * sshPublicKey: string
+ * pullSecretName: string (reference to k8s secret)
+ * ignitionOverrides: object
+ * staticIpConfiguration: object
+
+Status:
+ * state: string
+ * sizeBytes: integer
+ * downloadUrl: string
+ * createdAt: string
+ * expiresAt: string
+
+**AgentDrivenClusterInstallation**
+This resource correlates to the "cluster" resource in the assisted installer's
+current API.
+
+Spec:
+ * name: string
+ * approved: boolean (used for starting installation)
+ * openshiftVersion: string
+ * baseDnsDomain: string
+ * clusterNetworkCidr: string
+ * clusterNetworkHostPrefix: integer
+ * serviceNetworkCidr: string
+ * apiVip: string
+ * apiVipDnsName: string
+ * ingressVip: string
+ * machineNetworkCidr: string
+ * sshPublicKey: string
+ * vipDhcpAllocation: boolean
+ * httpProxy: string
+ * httpsProxy: string
+ * noProxy: string
+ * userManagedNetworking: boolean
+ * additionalNtpSource: string
+ * installConfigOverrides: string
+
+Status:
+* state: string enum
+* stateInfo: string
+* host_networks: array of (cidr, host_ids)
+* install_started_at: string
+* hosts: int
+* progress:
+  * progress_info: string
+  * progress_updated_at: string
+* validation_info:
+  * configuration:
+    * is_pull_secret_set: [success, failure, pending, error]
+  * host_data:
+    * all_hosts_are_ready_to_install: [success, failure, pending, error]
+    * sufficient_masters_count: [success, failure, pending, error]
+  * network:
+    * machine_cidr_defined: [success, failure, pending, error]
+    * cluster_cidr_defined: [success, failure, pending, error]
+    * service_cidr_defined: [success, failure, pending, error]
+    * no_cidrs_overlapping: [success, failure, pending, error]
+    * network_prefix_valid: [success, failure, pending, error]
+    * machine_cidr_equals_to_calculated_cidr: [success, failure, pending, error]
+    * api_vip_defined: [success, failure, pending, error]
+    * api_vip_valid: [success, failure, pending, error]
+    * ingress_vip_defined: [success, failure, pending, error]
+    * ingress_vip_valid: [success, failure, pending, error]
+    * dns_domain_defined: [success, failure, pending, error]
+    * ntp_server_configured: [success, failure, pending, error]
+* connectivity_majority_groups: string
+* updated_at: string
+* created_at: string
+* install_started_at: string
+* install_completed_at: string
+* controller_logs_collected_at: string
+
+**AgentDrivenHostInstallation**
+AgentDrivenHostInstallation represents a host that is destined to become part
+of an OpenShift cluster, and is running an agent that is able to run inspection
+and installation tasks. It correlates to the "host" resource in the Assisted
+Installer's REST API.
+
+Spec:
+ * clusterName: string
+ * role: [auto-assign, master, worker]
+ * hostname: string
+ * machineConfigPool: string
+ * disks:
+   * diskId: string
+   * role: [none, install]
+ * enabled: boolean
+ * ignitionConfigOverrides: string
+ * installerArgs: string
+
+Status:
+ * state: string enum
```

Comment:
should this be string alias, same as in line#172?

Reply:
@avishayt

Reply:
Yes